packages/core/src/App/Containers/Layout/header/traders-hub-header.tsx

```diff
@@ -0,0 +1,190 @@
+import classNames from 'classnames';
+import { useHistory, useLocation } from 'react-router-dom';
+import { DesktopWrapper, Icon, MobileWrapper, Popover, StaticUrl } from '@deriv/components';
+import { useIsRealAccountNeededForCashier } from '@deriv/hooks';
+import { routes, platforms, formatMoney } from '@deriv/shared';
+import { observer, useStore } from '@deriv/stores';
+import { Localize } from '@deriv/translations';
+import { MenuLinks } from 'App/Components/Layout/Header';
+import platform_config from 'App/Constants/platform-config';
+import ToggleMenuDrawer from 'App/Components/Layout/Header/toggle-menu-drawer.jsx';
+import { BinaryLink } from 'App/Components/Routes';
+import DerivBrandLogo from 'Assets/SvgComponents/header/deriv-rebranding-logo.svg';
+import RealAccountSignup from 'App/Containers/RealAccountSignup';
+import AccountInfo from 'App/Components/Layout/Header/account-info';
+import SetAccountCurrencyModal from 'App/Containers/SetAccountCurrencyModal';
+import CurrencySelectionModal from '../../CurrencySelectionModal';
+import DefaultMobileLinks from './default-mobile-links';
+import ShowNotifications from './show-notifications';
+import TradersHubOnboarding from './traders-hub-onboarding';
+import TradersHubHomeButton from './traders-hub-home-button';
+
+type TPlatformConfig = typeof platform_config;
+type TPlatforms = typeof platforms;
+
+const TradersHubHeader = observer(() => {
+    const { client, common, traders_hub, ui } = useStore();
+    const {
+        account_type,
+        balance,
+        country_standpoint,
+        currency,
+        has_any_real_account,
+        is_eu,
+        is_logged_in,
+        is_mt5_allowed,
+        is_virtual,
+    } = client;
+    const { platform } = common;
+    const { modal_data } = traders_hub;
+    const {
+        header_extension,
+        is_accounts_switcher_on,
+        is_app_disabled,
+        is_route_modal_on,
+        account_switcher_disabled_message,
+        toggleAccountsDialog,
+        toggleNeedRealAccountForCashierModal,
+        toggleReadyToDepositModal,
+    } = ui;
+    const { pathname } = useLocation();
+    const cashier_routes = pathname.startsWith(routes.cashier);
+    const history = useHistory();
+    const real_account_needed_for_cashier = useIsRealAccountNeededForCashier();
+
+    const filterPlatformsForClients = (payload: TPlatformConfig) =>
+        payload.filter(config => {
+            if (config.link_to === routes.mt5) {
+                return !is_logged_in || is_mt5_allowed;
+            }
+            return true;
+        });
+
+    const toggleModal = () => {
+        if (!has_any_real_account) {
+            toggleReadyToDepositModal();
+        } else if (window.location.pathname === routes.traders_hub) {
+            toggleNeedRealAccountForCashierModal();
+        }
+    };
+
+    const handleClickCashier = () => {
+        if ((!has_any_real_account && is_virtual) || real_account_needed_for_cashier) {
```

Comment:
Can the logic here be refactored in a better way to eliminate `toggleModal` function?

Reply:
clubbed it into single function. 👍🏼